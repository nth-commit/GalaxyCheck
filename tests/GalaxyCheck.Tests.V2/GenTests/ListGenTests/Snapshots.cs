using GalaxyCheck;
using GalaxyCheck.Gens;
using Snapshooter;
using Snapshooter.Xunit;
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Text.Json;
using System.Threading.Tasks;
using Xunit;

namespace Tests.V2.GenTests.ListGenTests
{
    public class Snapshots
    {
        [Fact]
        public void Values()
        {
            var seeds = Enumerable.Range(0, 5);

            var gens = new Dictionary<string, IListGen<int>>
            {
                { "Between_Lengths_1_And_4_And_Elements_0_And_10", Gen.Int32().Between(0, 10).ListOf().BetweenLengths(1, 4) },
                { "Between_Lengths_1_And_8_And_Elements_0_And_100", Gen.Int32().Between(0, 100).ListOf().BetweenLengths(1, 8) },
                { "Between_Lengths_1_And_16_And_Elements_0_And_10000", Gen.Int32().Between(0, 10000).ListOf().BetweenLengths(1, 16) },
                { "Between_Defaults", Gen.Int32().ListOf() },
            };

            var biases = new[] { Gen.Bias.None, Gen.Bias.WithSize };

            foreach (var seed in seeds)
            foreach (var (betweenDescription, gen) in gens)
            foreach (var bias in biases)
            {
                var gen0 = gen.WithLengthBias(bias);

                var sample = gen.Sample(seed: seed);

                var nameExtension = string.Join("_", new[]
                {
                    betweenDescription,
                    $"Bias_{bias}",
                    $"Seed_{seed}"
                });

                Snapshot.Match(sample, SnapshotNameExtension.Create(nameExtension));
            }
        }

        [Fact]
        public void ExampleSpaces()
        {
            var seeds = Enumerable.Range(0, 5);

            var gens = new Dictionary<string, IListGen<int>>
            {
                { "Between_Lengths_1_And_4_And_Elements_0_And_10", Gen.Int32().Between(0, 10).ListOf().BetweenLengths(1, 4) },
                { "Between_Lengths_1_And_8_And_Elements_0_And_100", Gen.Int32().Between(0, 100).ListOf().BetweenLengths(1, 8) },
            };

            var biases = new[] { Gen.Bias.None, Gen.Bias.WithSize };

            foreach (var seed in seeds)
            foreach (var (betweenDescription, gen) in gens)
            foreach (var bias in biases)
            {
                var gen0 = gen.WithLengthBias(bias);

                var sample = gen.Advanced
                    .SampleOneExampleSpace(seed: seed, size: 75)
                    .Take(500)
                    .Render(x => JsonSerializer.Serialize(x));

                var nameExtension = string.Join("_", new[]
                {
                    betweenDescription,
                    $"Bias_{bias}",
                    $"Seed_{seed}"
                });

                Snapshot.Match(sample, SnapshotNameExtension.Create(nameExtension));
            }
        }
    }
}
