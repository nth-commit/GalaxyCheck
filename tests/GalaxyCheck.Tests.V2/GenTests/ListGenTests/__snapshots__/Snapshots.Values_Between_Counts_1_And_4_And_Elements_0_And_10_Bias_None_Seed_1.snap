[
  [
    0
  ],
  [
    1
  ],
  [
    0
  ],
  [
    1,
    1
  ],
  [
    0
  ],
  [
    0,
    1
  ],
  [
    1,
    1
  ],
  [
    0
  ],
  [
    1,
    1
  ],
  [
    0
  ],
  [
    1,
    1
  ],
  [
    1
  ],
  [
    1,
    1
  ],
  [
    0
  ],
  [
    1
  ],
  [
    1,
    0
  ],
  [
    1
  ],
  [
    1,
    0
  ],
  [
    0,
    1
  ],
  [
    0,
    1
  ],
  [
    2,
    2,
    2,
    0
  ],
  [
    3,
    2
  ],
  [
    4
  ],
  [
    2,
    1,
    4
  ],
  [
    3
  ],
  [
    4,
    3,
    4,
    4
  ],
  [
    3,
    4
  ],
  [
    3,
    3,
    1,
    4
  ],
  [
    2
  ],
  [
    2,
    3,
    3,
    0
  ],
  [
    1,
    3
  ],
  [
    4,
    4,
    5,
    5
  ],
  [
    5,
    5,
    0,
    3
  ],
  [
    0
  ],
  [
    2,
    2,
    1,
    1
  ],
  [
    1,
    5,
    0
  ],
  [
    3
  ],
  [
    1,
    5,
    2,
    1
  ],
  [
    3,
    5,
    3
  ],
  [
    4,
    4
  ],
  [
    9,
    0,
    9,
    9
  ],
  [
    5,
    8,
    4
  ],
  [
    0,
    8,
    5
  ],
  [
    10
  ],
  [
    8,
    2,
    10
  ],
  [
    1
  ],
  [
    9,
    3,
    1,
    7
  ],
  [
    10
  ],
  [
    9,
    9,
    0
  ],
  [
    3,
    9
  ],
  [
    3
  ],
  [
    5,
    10
  ],
  [
    3,
    5,
    5
  ],
  [
    6,
    3
  ],
  [
    8
  ],
  [
    10,
    9,
    4
  ],
  [
    9,
    1
  ],
  [
    5,
    5,
    7
  ],
  [
    0,
    0,
    5,
    8
  ],
  [
    7,
    10
  ],
  [
    10,
    0
  ],
  [
    10,
    9,
    9,
    7
  ],
  [
    4,
    5
  ],
  [
    8
  ],
  [
    8,
    5,
    2
  ],
  [
    4,
    7,
    6
  ],
  [
    1,
    4
  ],
  [
    1,
    0,
    4,
    0
  ],
  [
    7,
    6,
    9
  ],
  [
    5,
    10,
    10
  ],
  [
    10,
    0,
    2,
    9
  ],
  [
    2,
    7,
    10
  ],
  [
    6
  ],
  [
    6,
    8
  ],
  [
    10,
    8,
    10,
    1
  ],
  [
    2
  ],
  [
    8,
    4
  ],
  [
    10,
    6
  ],
  [
    9,
    6
  ],
  [
    9
  ],
  [
    7,
    2,
    10,
    3
  ],
  [
    8,
    10,
    6,
    9
  ],
  [
    4
  ],
  [
    7,
    1,
    5,
    10
  ],
  [
    1,
    10,
    10,
    5
  ],
  [
    10
  ],
  [
    9
  ],
  [
    10,
    1,
    10,
    6
  ],
  [
    2,
    10,
    10
  ],
  [
    10,
    8,
    10,
    10
  ],
  [
    10,
    4,
    6
  ],
  [
    1
  ],
  [
    2
  ],
  [
    1,
    10
  ],
  [
    10,
    10,
    1,
    4
  ],
  [
    1,
    6,
    10,
    3
  ],
  [
    9,
    10,
    10,
    10
  ],
  [
    10,
    10,
    10,
    7
  ],
  [
    0,
    10,
    1,
    3
  ],
  [
    10
  ]
]
